src/main/java/org/opensearch/knn/index/codec/util/KNNCodecUtil.java

```diff
@@ -67,42 +67,44 @@ public static KNNCodecUtil.Pair getPair(final BinaryDocValues values, final Vect
         );
     }
 
+    /**
+     * This method provides a rough estimate of the number of bytes used for storing an array with the given parameters.
+     * @param numVectors number of vectors in the array
+     * @param vectorLength the length of each vector
+     * @param serializationMode serialization mode
+     * @return rough estimate of number of bytes used to store an array with the given parameters
+     */
     public static long calculateArraySize(int numVectors, int vectorLength, SerializationMode serializationMode) {
         if (serializationMode == SerializationMode.ARRAY) {
```

Comment:
It would be nice to add a comment on how this serialization mode store the vector in memory. Also please add a link to a reference where this formula came from.

If it is 2d array, float[][], the calculation here is not correct.
it should be something like(REFERENCE_SIZE or JAVA_ARRAY_HEADER_SIZE could be different between 32bit and 64bit)
```
int sizeOfVectorArray = vectorLength * FLOAT_BYTE_SIZE + JAVA_ARRAY_HEADER_SIZE;
sizeOfVectorArray = roundVectorSize(sizeOfVectorArray) * numVectors;

int sizeOfReferenceArray = numVectors * JAVA_REFERENCE_SIZE + JAVA_ARRAY_HEADER_SIZE;
sizeOfReferenceArray = roundVectorSize(sizeOfReferenceArray);

return sizeOfVectorArray + sizeOfReferenceArray;
```

Reply:
Yeah you're right, will change.  Forgot to add the reference on this recreated PR

Reply:
Hmm. You are right.. It was same :)

Reply:
Sorry. Let me look at this further. I still feel there is something incorrectness here..

Reply:
I looked at the code and this method mainly used to calculate how many off heap memory is being used during merge because we are allocating off heap memory to transfer vector data to JNI layer to construct a new graph. Then, I wonder why we need to distinguish between serialization mode. It is not about java memory or disk usage in serialized form but memory usage in pure float/byte array. Therefore, the memory size calculation should be simply `numVector * vectorLength * FLOAT_BYTE_SIZE` for float and `numVector * vectorLength` for byte.

Please let me know if I am understanding it incorrectly. @navneet1v, @martin-gaievski?
Original PR where this feature in introduced. https://github.com/opensearch-project/k-NN/pull/1141

Reply:
@navneet1v @martin-gaievski Specifically, I believe this is the conversation thread @heemin32 is referring to on that PR: https://github.com/opensearch-project/k-NN/pull/1141#discussion_r1336155264

Reply:
this is an interesting question. Now I am trying to think for what exact purpose we want these metrics. I have to think over it. Earlier when it was added because everything was in JVM so having JVM stats was good. Now with streaming change and iterative graph build what these metrics should represent. I don't have an exact ans here I have to think more on this.

Reply:
Okay now I am remember what it was for. I think we can remove the serialization mode and java things from the size calculation. Lets remove the java overhead. I agree with @heemin32  here.